include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1167 @@
+            $gModified = strtotime($event_remote->getUpdated());
+
+            // Get last modified of SuiteCRM event
+            $sModified = strtotime($event_local->fetched_row['date_modified'] . ' UTC'); // SuiteCRM stores the timedate as UTC in the DB
+
+            // Get the last sync time of SuiteCRM event
+            if (isset($event_local->fetched_row['gsync_lastsync'])) {
+                $lastSync = $event_local->fetched_row['gsync_lastsync'];
+            } else {
+                $lastSync = 0;
+            }
+
+            // Event has not been modified since last sync... skip
+            if ($gModified <= $lastSync && $sModified <= $lastSync) {
+                $this->syncedList[] = $event_local->id;
+                return "skip";
+            }
+
+            // Event was modified since last sync
+            if ($gModified > $lastSync || $sModified > $lastSync) {
+                if ($gModified > $sModified) {
+                    if ($event_remote->status == 'cancelled') {
+                         // if the remote event is deleted, delete it here
+                        return "pull_delete";
+                    } else {
+                        return "pull";
+                    }
+                } else {
+                    if ($event_local->deleted == '1') {
+                        return "push_delete";
+                    } else {
+                        return "push";
+                    }
+                }
+            }
+        }
+
+        return "unknown"; // we should never get here
+    }
+
+    /**
+     * Push event from SuiteCRM to Google Calendar
+     *
+     * If the google event is not provided, a new one will be created
+     * and inserted. If one is provided, the existing Google Event will
+     * be updated.
+     *
+     * @param Meeting $event_local : SuiteCRM Meeting Bean
+     * @param \Google_Service_Calendar_Event $event_remote (optional) \Google_Service_Calendar_Event Object
+     *
+     * @return bool Success/Failure
+     */
+    public function pushEvent($event_local, $event_remote = null)
+    {
+
+        if (!isset($event_remote) || empty($event_remote)) {
+            $event = $this->createGoogleCalendarEvent($event_local);
+            $return = $this->gService->events->insert($this->calendarId, $event);
+        } else {
+            $event = $this->updateGoogleCalendarEvent($event_local, $event_remote);
+            $return = $this->gService->events->update($this->calendarId, $event->getId(), $event);
+        }
+
+        // Set the SuiteCRM Meeting's last sync timestamp, and google id
+        $this->setLastSync($event_local, $return->getId());
+
+        /* We don't get a status code back showing success. Instead, the return of the
+         * create or update is the Google_Service_Calendar_Event object after saving. 
+         * So we check to make sure it has an ID to determine Success/Failure.
+         */
+        if (isset($return->id)) {
+            return true;
```

Comment:
This in itself is a workaround for the fact that the google api doesn't return a true/false. I was trying to keep things consistent with a success/failure return. Without this I need to put an isset check elsewhere. the Google ID is saved using the setLastSync call above this, which is the only consistent way to save a SuiteCRM record in the class. Otherwise the time stamp is not set and it doesn't have an accurate last sync to compare.

Reply:
in your case, if I calling your function in the future like this:
```
$ret = $g->pushEvent($x);
if ($ret) {
  echo "Ok, but what is the affected ID?";
} else {
  echo "Fail.";
}
```
if you returns something like this:
`return isset($return->id) ? $return->id : false;`
Then the caller knows and able to handle the information about the affected event.

Reply:
I set it to return the id instead of true.